examples/bloom/regressor.py

```diff
@@ -0,0 +1,124 @@
+"""Code for each data owner in the collaboration."""
+import numpy as np
+import tensorflow as tf
+import tf_encrypted as tfe
+
+
+class BloomRegressor:
+  """Secure multi-party linear regression at plaintext speed.
+
+  Computes the necessary components of the normal equations solution to a
+  linear regression."""
+  def __init__(self):
+    self.components = ["label_square",
+                       "covariate_label_product",
+                       "covariate_square"]
+
+  @classmethod
+  def estimator_fn(cls, x_p, y_p):
+    # Recall beta = np.inv(X.T @ X) * (X.T @ y)
+    yy_p = tf.matmul(y_p, y_p, transpose_a=True)  # per-party y.T @ y
+    xy_p = tf.matmul(x_p, y_p, transpose_a=True)  # per-party X.T @ y
+    xx_p = tf.matmul(x_p, x_p, transpose_a=True)  # per-party X.T @ X
+    return yy_p, xy_p, xx_p
+
+  def fit(self,
+          training_players,
+          summary=0,
+          validation_split=None):
+    if validation_split is not None:
+      raise NotImplementedError()
+
+    partial_estimators = [player.compute_estimators(self.estimator_fn)
+                          for player in training_players]
+
+    for attr, partial_estimator in zip(self.components,
+                                       zip(*partial_estimators)):
+      setattr(self, attr, tfe.add_n(partial_estimator))
+
+    with tfe.Session() as sess:
+      for k in self.components:
+        op = getattr(self, k)
+        setattr(self, k, sess.run(op.reveal()))
+
+    self._inverted_covariate_square = tf.linalg.inv(self.covariate_square)
```

Comment:
Creating a new graph should fix your device error:

```
tf_graph = tf.Graph()
with tf_graph.as_default():
      self._inverted_covariate_square = tf.linalg.inv(self.covariate_square)
      self.coefficients = tf.matmul(self._inverted_covariate_square,
                                    self.covariate_label_product)
```

Reply:
oof